src/app/wallet/v2/views/buy_sell_crypto/service_controller.nim

```diff
@@ -0,0 +1,56 @@
+import NimQml, json, strutils, chronicles
+
+import service_model, service_item
+
+import ../../../../../status/[status, wallet]
+
+logScope:
+  topics = "crypto-service"
+
+QtObject:
+  type CryptoServiceController* = ref object of QObject
+    status: Status
+    cryptoServiceModel: CryptoServiceModel 
+    servicesFetched: bool
+
+  proc setup(self: CryptoServiceController) = 
+    self.QObject.setup
+
+  proc delete*(self: CryptoServiceController) =
+    self.cryptoServiceModel.delete
+    self.QObject.delete    
+
+  proc newCryptoServiceController*(status: Status): CryptoServiceController =
+    new(result, delete)
+    result.status = status
+    result.cryptoServiceModel = newCryptoServiceModel()
+    result.servicesFetched = false
+    result.setup
+
+  proc getCryptoServiceModel(self: CryptoServiceController): QVariant {.slot.} = 
+    newQVariant(self.cryptoServiceModel)
```

Comment:
If you create a QObject without a parent provided, then QML context takes ownership of this object and may delete it when GC is running. Since this controller is clearly designed to be the owner, then you need to pass `parent`.
Unless NimQML has some magic I am not aware of.

Reply:
@pinebit yes, I know that, I faced that issue many years ago but that mostly happened to me if I was returning QObject raw pointer to in a method exposed to qml using `Q_INVOKABLE`. I was thinking about that some time ago how it figures in our app. This proc, you marked, is used for registering `READ` for `Q_PROPERTY` same way as we have on many places in the app, so maybe that should be done via some new issue, you can log that, we need to expose first `setObjectOwnership` method of `QQmlEngine` and then refactor all places in the app which are exposing raw pointer to qml.